### PR TITLE
Update report processor to add tag function based on Puppet facts.

### PR DESCRIPTION
**Pull request by murdok5**
Modify report class to manage report tags from facts through Datadog
config yaml file.

### What does this PR do?

This PR enables a new parameter to be set that specifies which Puppet facts will be added as tags to each report sent to DataDog as an event.

### Motivation

Working with a customer we found that correlating DataDog events and metrics was challenging and using Puppet Facts would simplify this problem.

### Additional Notes

This is outside my typical expertise and as such I would request detailed review/comments/opinions/testing from anyone.

### Describe your test plan

Using a current release of Datadog and Puppet Enterprise at the time of this PR I validated functionality and updated the readme with specific tips and note on how to use. I modified the values, deployed the Puppet code, restarted puppetserver service, and validated the new facts were appearing on the next Puppet run reports in Datadog.

### CODE REVIEW

File: templates/datadog-reports.yaml.erb [hunk 1]
```diff
@@ -12,3 +12,10 @@
 <% if @proxy_https -%>
 :proxy_https: <%= @proxy_https %>
 <% end -%>
+<% if @report_fact_tags -%>
+#:report_fact_tags: <%= @report_fact_tags %>
```
Comment (albertvaka) [inline templates/datadog-reports.yaml.erb:16]:
```suggestion
```

**Comment by murdok5** [comment]
thanks for the updates @cswatt and good catch @albertvaka

**Comment by albertvaka** [comment]
Merging then :) Thanks a ton for your patches! 🙇

**Comment by albertvaka** [comment]
@murdok5 I'm trying to extend this to add support for trusted facts. Is there something like `Puppet::Node::Facts.indirection.find(host).values` that I can use? Thanks!

**Pull request merged**